### PR TITLE
Input component

### PR DESCRIPTION
**Pull request by carlobernardini**
* Adds Input component

### CODE REVIEW

File: stories/Input.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
+import { boolean, select, text, withKnobs } from '@storybook/addon-knobs'; // eslint-disable-line import/no-extraneous-dependencies
+import Input from '../src/components/Input';
+import { CONTEXTS, INPUT_TYPES, SIZES } from '../src/constants';
+
+storiesOf('Input', module)
```
Comment (antipin) [inline stories/Input.js]: Would be nice to add a note like "Any other properties supplied will be spread to the underlined `<input />`  element"

File: src/constants/index.js [hunk 2]
```diff
@@ -1 +1,13 @@
...
+    'text',
+    'password',
+    'email',
+    'date',
+    'datetime-local',
+    'month',
+    'tel',
+    'number',
+    'url'
+];
```
Comment (antipin) [inline src/constants/index.js]: `search`, `time`, `week`, are missing

File: src/components/Input/Input.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        size: var(--font-size-normal);
+        weight: var(--font-weight-normal);
+    }
+    outline: none;
+    padding: var(--spacing-2x);
+    transition: border-color var(--transition-duration);
+
+    &:focus {
+        border-color: var(--color-neutral);
+        box-shadow: 0 0 0 3px var(--color-neutral-25);
```
Comment (antipin) [inline src/components/Input/Input.scss]: Oh my god! Dat is magic number! 🎃

File: src/components/Input/Input.scss [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        color: var(--color-neutral-75);
+        cursor: not-allowed;
+    }
+
+    &--context {
+        @each $context in (brand, primary, accent, info, good, warning, bad) {
+            &_#{ $context } {
+                &:focus {
+                    border-color: var(--color-#{ $context });
+                    box-shadow: 0 0 0 3px var(--color-#{ $context }-25);
```
Comment (antipin) [inline src/components/Input/Input.scss]: another one magic

File: src/components/Input/Input.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    );
+};
+
+Input.propTypes = {
+    /** The input field context (e.g. brand, primary, bad, good etc. - defaults to brand) */
+    context: PropTypes.oneOf(CONTEXTS),
+    /** Should the input field be disabled or not */
+    disabled: PropTypes.bool,
+    /** Callback function on change */
+    onChange: PropTypes.func,
```
Comment (antipin) [inline src/components/Input/Input.js]: Trying to come up with some solid idea about which event handlers should be listed in docs, and which can be passed silently by `{ ... rest }`. To list them all is stupid, to list only some of them can be confusing (like we support only that one), list none of them... don't know, but it's consistent.
@carlobernardini @ribeiroguilherme @eszthoff what do you think?

File: stories/Input.js (hunk 1 above)
Comment (carlobernardini) [reply stories/Input.js]: Can we maybe just add a code comment for now and add @storybook/addon-notes later in a new branch?

Comment (antipin) [reply stories/Input.js]: sure, just maybe worth to add a task about it in order not to miss it

File: src/components/Input/Input.js (hunk 5 above)
Comment (carlobernardini) [reply src/components/Input/Input.js]: I think you're right. But if we don't define any of them, none of them will have any validation. Defining all of them isn't a good idea and would take a lot of work.
I can't think of a better solution than making it very clear through docs (as I suggest above we can also add @storybook/addon-notes for extra notes in Storybook)

Comment (antipin) [reply src/components/Input/Input.js]: > But if we don't define any of them, none of them will have any validation.

You mean propTypes validation? But do we really need to manage it? As far as we don't add any extra code around `onChange` callback, we don't need to validate it I guess. Or I don't get your point

Comment (carlobernardini) [reply src/components/Input/Input.js]: Yes I meant propTypes validation.

For the record, we decided not to explicitly define props like `onChange` or `onClick` but just propagate all of these through `{...rest}`. Will update `<Button />` as well.

**Pull request merged**